### PR TITLE
fix(react-textarea): Remove fixed height so rows prop works and add missing max-height

### LINKED ISSUE #27842: [Bug]: TextArea rows not working
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
NA
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/nameless-meadow-cwkc6x?file=/example.tsx:48-74

### Bug Description

Settings rows in Textarea does not increase element size.
Workaround can be done by setting  textarea={{style:{height:"auto"}}} 

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

`<Textarea rows={6} />` hides the added rows.
image

## New Behavior

`<Textarea rows={6} />` shows the added rows.
image

## Related Issue(s)

- Fixes #27842

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProvider-mount_base.html">646</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProvider-mount.html">647</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProvider-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/InfoButton-mount_base.html">11</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Avatar-mount_base.html">582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Avatar-mount.html">567</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Button-mount_base.html">302</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Button-mount.html">281</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Field-mount_base.html">1006</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Field-mount.html">1050</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProvider-mount_base.html">646</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProvider-mount.html">647</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProvider-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/InfoButton-mount_base.html">11</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/MakeStyles-mount_base.html">861</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/MakeStyles-mount.html">849</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Persona-mount_base.html">1571</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/Persona-mount.html">1538</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/SpinButton-mount_base.html">1169</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28181/perf-test-react-components/SpinButton-mount.html">1236</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                |    Baseline (minified/GZIP) |                         PR |                                                                                                                                                                                                   Change |
| :----------------------------------------------------------------------------------------------- | --------------------------: | -------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr> | `27.722 kB`<br />`9.135 kB` | `27.69 kB`<br />`9.125 kB` | `-32 B` <br />`-10 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `64.935 kB`<br />`17.918 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.151 kB`<br />`58.715 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.168 kB`<br />`11.964 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/91b9ce05995846aaa882d86755583e5b1a703717'>91b9ce05995846aaa882d86755583e5b1a703717</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts [hunk 1]
```diff
@@ -188,30 +181,30 @@ const useTextareaStyles = makeStyles({
   // The padding style adds both content and regular padding (from design spec), this is because the handle is not
   // affected by changing the padding of the root.
   small: {
-    height: textareaHeight.small,
     minHeight: '40px',
     ...shorthands.padding(
       tokens.spacingVerticalXS,
       `calc(${tokens.spacingHorizontalSNudge} + ${tokens.spacingHorizontalXXS})`,
     ),
+    maxHeight: '200px',
```
Comment (khmakoto) [inline packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts:189]: Are these values specified by design? Where are they coming from?

Comment (sopranopillow) [reply packages/react-components/react-textarea/src/components/Textarea/useTextareaStyles.styles.ts:189]: Yes, they come from the figma spec

**Review by khmakoto — APPROVED** [review]

**Pull request merged**